features/lorawan/lorastack/phy/LoRaPHY.h

```diff
@@ -591,6 +628,70 @@ class LoRaPHY : private mbed::NonCopyable<LoRaPHY> {
      */
     uint32_t get_rejoin_max_count() const;
 
+    /*!
+     * @brief Computes beacon RX window frequency, timeout and offset.
+     *
+     * @param [in] beacon_time      Beacon time for frequency calculation
+     *
+     * @param [in] min_rx_symbols   The minimum number of symbols required to
+     *                              detect an RX frame.
+     *
+     * @param [in] rx_error         The maximum timing error of the receiver
+     *                              in milliseconds. The receiver will turn on
+     *                              in a [-rxError : +rxError] ms interval around
+     *                              RxOffset.
+     *
+     * @param [out] rx_conf_params  Pointer to the structure that needs to be
+     *                              filled with receive window parameters.
+     *
+     * @return False, if invalid receive parameters
+     */
+    virtual bool compute_beacon_win_params(uint32_t beacon_time, uint8_t min_rx_symbols,
+                                           uint32_t rx_error, rx_config_params_t *config);
+
+    /**
+     * @brief get_beacon_rfu_size Gets the current region beacon RFU field sizes
+     */
+    virtual void get_beacon_rfu_size(uint8_t &rfu1, uint8_t &rfu2);
+
+    /**
+     * @brief get_beacon_frequency Computes beacon frequency
+     * @param beacon_time beacon time used to compute beacon frequency
+     * @return beacon frequency
+     */
+    virtual uint32_t get_beacon_frequency(uint32_t beacon_time);
+
+    /*!
+     * Computes the ping slot frequency, window timeout and offset.
+     *
+     * @param [in] beacon_time      The current beacon period time
+     *
+     * @param [in] devaddr          Device address for ping frequency computation
+     *
+     * @param [in] min_rx_symbols   The minimum number of symbols required to
+     *                              detect an RX frame.
+     *
+     * @param [in] rx_error         The maximum timing error of the receiver
+     *                              in milliseconds. The receiver will turn on
+     *                              in a [-rxError : +rxError] ms interval around
+     *                              RxOffset.
+     *
+     * @param [out] rx_conf_params  Pointer to the structure that needs to be
+     *                              filled with receive window parameters.
+     *
+     * @return False, if invalid receive parameters
+     */
+    virtual bool compute_ping_win_params(uint32_t beacon_time, uint32_t devaddr,
+                                         uint8_t min_rx_symbols, uint32_t rx_error,
+                                         rx_config_params_t *config);
+
+    /**
+     * @brief compute_beacon_time_on_air
+     * @param compute_beacon_time_on_air computes beacon frame time on air
```

Comment:
param - there's no parameter, brief should contain this description?

Reply:
Fixed